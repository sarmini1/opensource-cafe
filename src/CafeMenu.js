import React from "react";
import { Link } from "react-router-dom";
import "./CafeMenu.css";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  ListGroup,
  ListGroupItem
} from "reactstrap";

/**CafeMenu
 *
 * Props:
 * - type: string
 * - items: array of objects [{id, name, description, recipe, serve}...]
 * - title: string
 *
 * State:
 * - none
 *
 * App --> NavBar, Routes --> CafeMenu
 */
function CafeMenu({ type, items, title }) {

  return (
    <section className="CafeMenu col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            {title} Menu
          </CardTitle>
          <CardText>
            Please enjoy some of our {type} listed below!
          </CardText>
          <ListGroup>
            {items.map(item => (
              <Link to={`/${type}/${item.id}`} key={item.id}>
                <ListGroupItem>{item.name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
        </CardBody>
      </Card>
    </section>
  );
}

export default CafeMenu;