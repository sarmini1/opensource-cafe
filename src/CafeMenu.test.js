import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";
import CafeMenu from "./CafeMenu";

const testSnacks = [
  {
    "id": "nachos",
    "name": "Nachos",
    "description": "An American classic!",
    "recipe": "Cover expensive, organic tortilla chips with Cheez Whiz.",
    "serve": "Serve in a hand-thrown ceramic bowl, garnished with canned black olives"
  },
  {
    "id": "hummus",
    "name": "Hummus",
    "description": "Sure to impress your vegan friends!",
    "recipe": "Purchase one container of hummus.",
    "serve": "Place unceremoniously on the table, along with pita bread."
  }];

  it('mounts without crashing', function () {
    render(<MemoryRouter><CafeMenu type={"snacks"} items={testSnacks} title={"Snacks"}/></MemoryRouter>);
  });
  
  it("matches snapshot", function () {
    const {debug, container} = render(<MemoryRouter><CafeMenu type="snacks" items={testSnacks} title="Snacks"/></MemoryRouter>);
    debug(container);
    expect(container).toMatchSnapshot();
  });

  it("displays a menu of items", function (){
    const {debug, container} = render(<MemoryRouter><CafeMenu type="snacks" items={testSnacks} title="Snacks"/></MemoryRouter>);

    debug(container);
    expect(container).toContainHTML("Hummus");
    expect(container).toContainHTML("Nachos");
  });