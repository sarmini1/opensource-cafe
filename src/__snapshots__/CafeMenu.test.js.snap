// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div
          class="font-weight-bold text-center card-title"
        >
          Snacks
           Menu
        </div>
        <p
          class="card-text"
        >
          Please enjoy some of our 
          snacks
           listed below!
        </p>
        <ul
          class="list-group"
        >
          <a
            href="/snacks/nachos"
          >
            <li
              class="list-group-item"
            >
              Nachos
            </li>
          </a>
          <a
            href="/snacks/hummus"
          >
            <li
              class="list-group-item"
            >
              Hummus
            </li>
          </a>
        </ul>
      </div>
    </div>
  </section>
</div>
`;
